test/unit.js

```diff
@@ -57,6 +57,78 @@ var tests = [
   {
     fixture: "e1",
     expected: false
+  },
+  {
+    fixture: "2rem",
+    expected: { number: "2", unit: "rem" }
+  },
+  {
+    fixture: "2.000rem",
+    expected: { number: "2.000", unit: "rem" }
+  },
+  {
+    fixture: "+2rem",
+    expected: { number: "+2", unit: "rem" }
+  },
+  {
+    fixture: "-2rem",
+    expected: { number: "-2", unit: "rem" }
+  },
+  {
+    fixture: "1.1rem",
+    expected: { number: "1.1", unit: "rem" }
+  },
+  {
+    fixture: "+1.1rem",
+    expected: { number: "+1.1", unit: "rem" }
+  },
+  {
+    fixture: "-1.1rem",
+    expected: { number: "-1.1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e1rem",
+    expected: { number: "1.1e1", unit: "rem" }
+  },
+  {
+    fixture: "+1.1e1rem",
+    expected: { number: "+1.1e1", unit: "rem" }
+  },
+  {
+    fixture: "-1.1e1rem",
+    expected: { number: "-1.1e1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e+1rem",
+    expected: { number: "1.1e+1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e-1rem",
+    expected: { number: "1.1e-1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e1e1rem",
+    expected: { number: "1.1e1", unit: "e1rem" }
+  },
+  {
+    fixture: "1.1e-1e",
+    expected: { number: "1.1e-1", unit: "e" }
+  },
+  {
+    fixture: "1.1e-1rem",
+    expected: { number: "1.1e-1", unit: "rem" }
+  },
+  {
+    fixture: "1.1e--++1e",
+    expected: { number: "1.1e-", unit: "-++1e" }
```

Comment:
Is this valid number?

Reply:
@TrySound no, we already stop parsing for second `e`, so i use same logic for `+` and `-`

Reply:
Our fixtures contains:
```
{
    fixture: "2.",
    expected: { number: "2.", unit: "" }
  },
```

`2.` is not valid number too :smile: